WordPress/gutenberg
# Fix sample code on InputControl docs

## What?
InputControl code samples broken - the `useState` package has moved from `wordpress/compose` to `wordpress/element`

## How?
When trying to use an InputControl, I found that many of the Gutenberg package imports had been updated, but not those for the InputControl component. This affects the [Developer Documentation](https://developer.wordpress.org/block-editor/reference-guides/components/input-control/#usage) and also [Storybook](https://wordpress.github.io/gutenberg/?path=/docs/components-experimental-inputcontrol--docs). 

There were three declarations that needed to be changed in the Gutenberg folder. Search & Replace in VS Code worked to fix the code and this PR represents the updated Gutenberg codebase with changes to the developer documentation. 

## Testing Instructions

1. `npx @wordpress/create-block input-control-block`
2. Start the build to watch for changes
3. Activate the Input Control Block plugin
4. Add an input control block to a new post
5. Modify the edit function with the [sample code](https://developer.wordpress.org/block-editor/reference-guides/components/input-control/#usage), see steps below
6. Add the import headers above the Edit() function
```js
import { __experimentalInputControl as InputControl } from '@wordpress/components';
import { useState } from '@wordpress/compose';
```
7. Change the Edit function to match sample code
```js 
export default function Edit() {
	const [ value, setValue ] = useState( '' );
	
	return (
		<InputControl
			value={ value }
			onChange={ ( nextValue ) => setValue( nextValue ?? '' ) }
		/>
	);
};
```

8. At this point the block breaks
<img width="718" alt="Screenshot 2024-03-01 at 5 10 47 PM" src="https://github.com/WordPress/gutenberg/assets/3792502/a8af40bb-106c-49e5-8f92-429b9c84a5f4">
console error shown below

<img width="862" alt="Screenshot 2024-03-01 at 5 11 16 PM" src="https://github.com/WordPress/gutenberg/assets/3792502/3d558c9f-624c-4a8a-b5bd-d4059882b396">


### Testing the broken interface

1. In Edit.js
**change**
```js
import { useState } from '@wordpress/compose';
```
**to**
```js
import { useState } from '@wordpress/element';
```
3. Save changes
4. Refresh the page or post where you inserted the block
6. The block should now show an input box
<img width="703" alt="Screenshot 2024-03-01 at 5 25 18 PM" src="https://github.com/WordPress/gutenberg/assets/3792502/f775c916-a4a1-45d0-a599-73de4256f435">

## About this PR
This PR simply changes the location of the package imports, so no one else runs into this when trying to work with the InputControl component.

### Screenshots included
- UI when not working
- Error code
- Functional UI
